x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/huggingface/rerank/HuggingFaceRerankServiceSettings.java

```diff
@@ -0,0 +1,154 @@
+import org.elasticsearch.xpack.inference.services.huggingface.HuggingFaceRateLimitServiceSettings;
+import org.elasticsearch.xpack.inference.services.huggingface.HuggingFaceService;
+import org.elasticsearch.xpack.inference.services.settings.FilteredXContentObject;
+import org.elasticsearch.xpack.inference.services.settings.RateLimitSettings;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.elasticsearch.xpack.inference.services.ServiceFields.MAX_INPUT_TOKENS;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.createUri;
+import static org.elasticsearch.xpack.inference.services.huggingface.HuggingFaceServiceSettings.extractUri;
+
+public class HuggingFaceRerankServiceSettings extends FilteredXContentObject
+    implements
+        ServiceSettings,
+        HuggingFaceRateLimitServiceSettings {
+
+    public static final String NAME = "hugging_face_rerank_service_settings";
+    public static final String URL = "url";
+    private static final int RERANK_TOKEN_LIMIT = 512;
+
+    private static final RateLimitSettings DEFAULT_RATE_LIMIT_SETTINGS = new RateLimitSettings(3000);
+
+    public static HuggingFaceRerankServiceSettings fromMap(Map<String, Object> map, ConfigurationParseContext context) {
+        ValidationException validationException = new ValidationException();
+        var uri = extractUri(map, URL, validationException);
+        RateLimitSettings rateLimitSettings = RateLimitSettings.of(
+            map,
+            DEFAULT_RATE_LIMIT_SETTINGS,
+            validationException,
+            HuggingFaceService.NAME,
+            context
+        );
+
+        if (validationException.validationErrors().isEmpty() == false) {
+            throw validationException;
+        }
+        return new HuggingFaceRerankServiceSettings(uri, rateLimitSettings);
+    }
+
+    private final URI uri;
+    private final RateLimitSettings rateLimitSettings;
+
+    public HuggingFaceRerankServiceSettings(String url) {
+        uri = createUri(url);
+        rateLimitSettings = DEFAULT_RATE_LIMIT_SETTINGS;
+    }
+
+    HuggingFaceRerankServiceSettings(URI uri, @Nullable RateLimitSettings rateLimitSettings) {
+        this.uri = Objects.requireNonNull(uri);
+        this.rateLimitSettings = Objects.requireNonNullElse(rateLimitSettings, DEFAULT_RATE_LIMIT_SETTINGS);
+    }
+
+    public HuggingFaceRerankServiceSettings(StreamInput in) throws IOException {
+        uri = createUri(in.readString());
+
+        if (in.getTransportVersion().onOrAfter(TransportVersions.V_8_15_0)) {
+            rateLimitSettings = new RateLimitSettings(in);
+        } else {
+            rateLimitSettings = DEFAULT_RATE_LIMIT_SETTINGS;
+        }
+    }
+
+    @Override
+    public RateLimitSettings rateLimitSettings() {
+        return rateLimitSettings;
+    }
+
+    @Override
+    public URI uri() {
+        return uri;
+    }
+
+    public int maxInputTokens() {
```

Comment:
As I noted about I don't think we'll need these since they're really only used for text embedding style task types.